### PR TITLE
Revenue management

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#TicketID](https://edu-redmine.sun-asterisk.vn/issues/45411)

## WHAT (optional)
- statistics users and selling products

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image

### CODE REVIEW

**Review by hoangtv-2888 — APPROVED** [review]

**Comment by ghost** [comment]
ready

File: app/controllers/admin/revenue_managements_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  before_action :load_orders, :load_users
+  def index; end
+
+  private
+  def handle_err obj
+    return if obj.present?
+
+    flash[:warning] = t "not_found"
+    redirect_to admin_root_path
+  end
```
Comment (manhnt-0960) [inline app/controllers/admin/revenue_managements_controller.rb]: cái này nên để ngoài admin controller hoặc application controller để các controller khác dùng nữa

File: app/models/order_detail.rb [hunk 2]
```diff
@@ -4,4 +4,8 @@ class OrderDetail < ApplicationRecord
 
   validates :quantity, presence: true
   validates :price_at_order, presence: true
+
+  scope :count_order, (lambda do |status|
```
Comment (manhnt-0960) [inline app/models/order_detail.rb]: tên scope nên là by_order_status

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
+  namespace :admin do
+    get 'revenue_managements/index'
+  end
+  get 'revenue_managements/index'
```
Comment (manhnt-0960) [inline config/routes.rb]: thừa à e

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**